frontend/src/components/elements/DataFrame/DataFrameUtil.tsx

```diff
@@ -99,15 +101,20 @@ export const getDimensions = (
   const headerHeight = rowHeight * headerRows
   const border = 2
 
-  let { elementWidth, columnWidth, headerWidth } = getWidths(
+  const widths = getWidths(
     cols,
     rows,
     headerCols,
     headerRows,
-    width - border,
+    // Reserve enough space to render a vertical scrollbar in case we need to
+    // do so.
+    width - border - scrollbarSize(),
```

Comment:
Nit: extract this to a new variable, for clarity?

`const containerWidth = width - border - scrollbarSize()`